File: cplusplus/array/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numTracker;
        vector<int> res;

        for (int i = 0; i < nums.size(); i++) {
            int ntf = target - nums[i];

            if (numTracker.contains(ntf)) {
                res.push_back(i);
                res.push_back(numTracker.at(ntf));
                return res;
            }
            else {
                numTracker.insert({ nums[i], i });
            }
        }
        return {};
    }
};

//https://leetcode.com/problems/two-sum

File: python/array/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        tracker = {}
        for i, v in enumerate(nums):
            ntf = target - v
            if ntf in tracker:
                return [i, tracker.get(ntf)]
            else:
                tracker[v] = i